connection.go

```diff
@@ -64,14 +64,24 @@ func (mc *mysqlConn) handleParams() (err error) {
 				return
 			}
 
-		// Other system vars
+		// Other system vars accumulated in a single SET command
 		default:
-			params = append(params, param+"="+val)
+			if cmdSet.Len() == 0 {
+				// Heuristic: 29 chars for each other key=value to reduce reallocations
+				cmdSet.Grow(4 + len(param) + 1 + len(val) + 30*(len(mc.cfg.Params)-1))
+				cmdSet.WriteString("SET ")
+			} else {
+				cmdSet.Grow(1 + len(param) + 1 + len(val))
```

Comment:
I feel this is too much.

Reply:
Why too much ? Too verbose ?

This code is executed every time a connection is opened.

Reply:
Builder.Grow() does overallocation to reduce reallocation.
So calling `cmdSet.Grow(...)` on every param won't reduce allocation.

Reply:
Fine. I just removed that `Grow`.